epuck_ros2_camera/src/camera.cpp

```diff
@@ -0,0 +1,175 @@
+
+    publisher_compressed = this->create_publisher<sensor_msgs::msg::CompressedImage>(
+        "image_raw/compressed", 0);
+    publisher_raw = this->create_publisher<sensor_msgs::msg::Image>(
+        "image_raw", 0);
+    timer = this->create_wall_timer(
+        std::chrono::milliseconds(interval), std::bind(&CameraPublisher::timer_callback, this));
+  }
+
+  ~CameraPublisher()
+  {
+    if (v4l2_initialized == true) {
+      pipuck_v4l2_deinit();
+    }
+    if (jpeg_initialized == true) {
+      pipuck_jpeg_deinit();
+    }
+  }
+
+private:
+  rcl_interfaces::msg::SetParametersResult param_change_callback(std::vector<rclcpp::Parameter> parameters)
+  {
+    auto result = rcl_interfaces::msg::SetParametersResult();
+    result.successful = true;
+
+    for (auto parameter : parameters)
+    {
+      if (parameter.get_name() == "quality")
+      {
+        if (jpeg_initialized == true)
+        {
+          pipuck_jpeg_deinit();
+        }
+        compressed_image.quality = parameter.as_int();
+        if (jpeg_initialized == true)
+        {
+          pipuck_jpeg_init(&captured_image, &compressed_image);
+        }
+      }
+      else if (parameter.get_name() == "interval")
+      {
+        timer->cancel();
+        timer = this->create_wall_timer(
+            std::chrono::milliseconds(parameter.as_int()), std::bind(&CameraPublisher::timer_callback, this));
+      }
+
+      RCLCPP_INFO(
+          this->get_logger(),
+          "Parameter '%s' has changed to %s",
+          parameter.get_name().c_str(),
+          parameter.value_to_string().c_str());
+    }
+
+    return result;
+  }
+
+  void timer_callback()
+  {
+    if (publisher_compressed->get_subscription_count() > 0 || publisher_raw->get_subscription_count())
+    {
+      if (v4l2_initialized == false)
+      {
+        pipuck_v4l2_init();
+        v4l2_initialized = true;
+      }
+      pipuck_v4l2_capture(&captured_image);
+    }
+    else
+    {
+      if (v4l2_initialized == true)
+      {
+        pipuck_v4l2_deinit();
+        v4l2_initialized = false;
+      }
+    }
+
+    if (publisher_raw->get_subscription_count() > 0)
+    {
+      cv::Mat input_mat(captured_image.height, captured_image.width, CV_8UC2, captured_image.data);
+      cv::Mat output_mat;
+      // COLOR_YUV2BGR_Y422
+      cv::cvtColor(input_mat, output_mat, cv::COLOR_YUV2BGR_YUY2); 
+
+      auto message = sensor_msgs::msg::Image();
+      message.encoding = "rgb8";
+      message.width = captured_image.width;
+      message.height = captured_image.height;
+      message.step = captured_image.width * 3;
+      message.is_bigendian = false;
+      message.header.stamp = now();
+      message.header.frame_id = "pipuck_image_raw";
+      message.data.assign(output_mat.data, output_mat.data + captured_image.height * captured_image.width * 3);
+    }
+
+    if (publisher_compressed->get_subscription_count() > 0)
+    {
+      if (jpeg_initialized == false)
```

Comment:
We generally avoid testing boolean expressions against true or false: https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs351-do-not-test-boolean-expressions-against-true-or-false